### ISSUE TITLE
What's the best approach to run actor service inside a container?

### ISSUE THREAD

**Issue by tomkerkhove** [dapr/dotnet-sdk#354]
What's the best approach to run actor service inside a container? Is there some guidance on how to approach this?

From what I'm seeing you can either use the CLI or Kubernetes ([docs](https://github.com/dapr/docs/tree/master/howto/invoke-and-discover-services#setup-an-id-using-kubernetes)) but what if we want to run it on Docker Compose with Visual Studio?

**Comment by amanbha** [comment]
@tomkerkhove Please see this sample to running app with docker compose https://github.com/dapr/samples/tree/master/10.hello-docker-compose

**Comment by tomkerkhove** [comment]
That was helpful, thanks!

**Comment by tomkerkhove** [comment]
Would it be useful if I send a PR to extend this sample with a demo?
https://github.com/dapr/dotnet-sdk/tree/master/samples/Actor

**Comment by amanbha** [comment]
@tomkerkhove yes, that would be very helpful, thanks!

**Comment by akourbat** [comment]
I assume the sample moved to here https://github.com/dapr/dotnet-sdk/tree/master/examples/Actor, but it does not contain any hints on how to write Docker-Compose to expose a containerized actor service in a multi-container apps. In particular, how to deal with --dapr-http-port 3500 in dockerfiles

**Comment by tomkerkhove** [comment]
This is what I'm using:
```yaml
version: '3.4'

services:
  app:
    image: app
    build:
      context: .
      dockerfile: app/Dockerfile
      args: 
      - VERSION=1.0.0
    ports:
      - "885:99"
      - "3000"
    depends_on:
      - dapr.statestores.redis
      - dapr.service.placement
    networks:
      - dapr

  app.sidecars.dapr:
    container_name: app.sidecars.dapr
    image: "daprio/daprd:edge"
    command: ["./daprd",
      "-app-id", "myapp",
      "-app-port", "3000",
      "-placement-host-address", "dapr.service.placement:50006",
      "-components-path", "/components"]
    volumes:
      - "./Dapr/:/components"
    depends_on:
      - app
    network_mode: "service:app"

  dapr.service.placement:
    container_name: dapr.statestores.placement
    image: "daprio/dapr"
    command: ["./placement", "-port", "50006"]
    ports:
      - "50006:50006"
    networks:
      - dapr

  dapr.statestores.redis:
    container_name: dapr.statestores.redis
    image: "redis:alpine"
    ports:
      - "6380:6379"
    networks:
      - dapr

networks:
  dapr:
    driver: bridge

```

**Comment by akourbat** [comment]
Interesting, thanks, I'll try that.  Tbh, I did not expect to have to manually specify placement and redis, as they are already running independently of my app after dapr init

**Comment by tomkerkhove** [comment]
Because you have to have the runtime run somewhere? Maybe dapr run handles this for you but I have never used that before.

**Comment by akourbat** [comment]
Alas, could not make it work even on your example.  Here's my config (I have 2 asp core apps, backend being the one with actors.  As soon as I put then on Dapr network, even direct method invocations start throwing, not just call via actor proxies.  I'll search a bit more on the web what could be the issue, else I'd have to back to Orleans and serverless actors... `version: '3.4'
```
version: '3.4'

services:
  daprfrontend:
    image: ${DOCKER_REGISTRY-}daprfrontend
    build:
      context: .
      dockerfile: DaprFrontEnd/Dockerfile
    ports:
      - "51000:50001"
    #networks:
    #  - dapr

  daprfrontend-dapr:
    image: "daprio/daprd:latest"
    command: [ "./daprd", "-app-id", "daprfrontend", "-app-port", "80" ]
    depends_on:
      - daprfrontend
    network_mode: "service:daprfrontend"

  daprbackend:
    image: ${DOCKER_REGISTRY-}daprbackend
    build:
      context: .
      dockerfile: DaprBackEnd/Dockerfile
    depends_on:
      - dapr.statestores.redis
      - dapr.service.placement
    ports:
      - "52000:50001"
    #networks:
    #  - dapr

  daprbackend-dapr:
    image: "daprio/daprd:latest"
    command: [ "./daprd",
      "-app-id", "daprbackend",
      "-app-port", "80",
      "-placement-host-address", "dapr.service.placement:50006",
      "-components-path", "/components" ]
    volumes:
      - "./Dapr/:/components"
    depends_on:
      - daprbackend
    network_mode: "service:daprbackend"

  dapr.service.placement:
    container_name: dapr.statestores.placement
    image: "daprio/dapr"
    command: [ "./placement", "-port", "50006" ]
    ports:
      - "50006:50006"
    networks:
      - dapr

  dapr.statestores.redis:
    container_name: dapr.statestores.redis
    image: "redis:alpine"
    ports:
      - "6380:6379"
    networks:
      - dapr

networks:
  dapr:
    driver: bridge
```

**Comment by tomkerkhove** [comment]
The app ports that you are adding are not defined on the `ports` section of the application containers so it cannot reach the app itself. Try adding `- "80"` to them.

However, also make sure the ports are not conflicting so you'll need to use a different app port for the other app & sidecar and make sure they are all part of the network.

**Comment by akourbat** [comment]
I actually resolved my issues, some of them were indeed port conflicts, but main mistakes were using `localhost` in the redis component file instead of actual name for my redis service from `docker-compose`, and somehow I did not include `Dapr.Actors` nuget in the actor service itself.  I guess I assumed that `Dapr.Actors.AspNetCore` would take care of bringing it in, but it does not.  Thanks for you help, nice to have stuff working.
